extensions/migrate.md

```diff
@@ -0,0 +1,321 @@
+# IRCv3 `migrate` Extension
+
+*Copyright &copy; 2016 Alex Iadicicco <https://github.com/aji>*
+
+*Copyright &copy; 2016 Justin Kaufman <https://github.com/akaritakai>*
+
+*Unlimited redistribution and modification of this document is allowed provided
+that the above copyright notice and this permission notice remains intact*
+
+This document describes the `migrate` IRCv3 client capability and associated
+protocol, which allows servers to migrate client sessions between servers, with
+appropriate client cooperation.
+
+Clients that support the migration protocol outlined in this document should
+register the `migrate` client capability. If the server supports client
+migrations (even if the network is not in a configuration to support
+migrations), it should accept the capability.
+
+# Protocol overview
+
+From the client's perspective, a migration works as follows. The details of this
+protocol are described later in this section.
+
+1. The server sends `MIGRATE START` to indicate that it wants to migrate the
+   client to a different server. Included in this message are a *resume token*
+   and *confirm token* that will be used when connecting to the new server.
+
+2. The client confirms the migration request with `MIGRATE OK`. After sending
+   `MIGRATE OK` the client should stop sending normal IRC traffic, as it will
+   not be processed. If for some reason the client does not wish to migrate,
+   then the migration fails. After receiving `MIGRATE OK`, the server can begin
```

Comment:
Should it reply explicitely it does not want to migrate? (eg. `MIGRATE NAK`)

Reply:
I'm not sure how it would "fail" otherwise, other than that it just wouldn't happen and the server would keep waiting on it.